### ISSUE TITLE
Column-stage fast path is never taken when `target_aware=True'

### ISSUE THREAD

**Issue by gcskoenig** [soda-inria/tabicl#155]
While reimplementing the TabICL v2 column stage I noticed that the `skip_value` fast path for the CLS columns is never taken when `target_aware=True`.

CLS columns are initialized with a skip value (`-100`), signalling that for this column the attention computation can be skipped. When `target_aware=True`, this skip value is overwritten, and the fast path is never taken. This does not affect the output of the model, only its computational efficiency.

A Claude-generated, more detailed description of the issue can be found below. I will open a PR with a fix.

## Summary by Claude

`ColEmbedding._compute_embeddings` (`_model/embedding.py:372-390`) does:

```python
src = self.in_linear(features)                                 # SkippableLinear -> reserved slots are exactly -100
...
src[..., :train_size, :] = src[..., :train_size, :] + y_emb    # embedding.py:389
src = self.tf_col(src, train_size=...)
```

and the guard it feeds is (`_model/layers.py:673`):

```python
skip_mask = (src == self.skip_value).all(dim=(-2, -1))
```

The guard requires *every* row and channel of a slice to equal `-100.0`. Adding `y_emb` to the first `train_size` rows breaks that for every reserved slot, so `skip_mask.any()` is `False` from line 389 onward and the fast path is dead.

The `not self.target_aware` branch three lines earlier calls `tf_col` directly after `in_linear`, where the guard works as intended — so this looks like the target-aware path having invalidated an optimisation written for the other one, rather than dead code.

### Evidence

Measured by wrapping `InducedSelfAttentionBlock.forward` to count how often its skip guard is true, with `ColEmbedding` built at the released v2 config and random weights (the branch taken does not depend on trained weights), `T=1000`, `train_size=800`, tabicl 2.2.0 on CPU:

```
features  fired (shipped)  fired (fixed)   shipped    fixed  speedup  feature slots identical
       5              0/3            3/3    0.027s   0.018s    1.53x                     True
      10              0/3            3/3    0.041s   0.031s    1.30x                     True
      25              0/3            3/3    0.083s   0.074s    1.13x                     True
      50              0/3            3/3    0.155s   0.149s    1.04x                     True
     100              0/3            3/3    0.296s   0.293s    1.01x                     True
```

`fired` counts ISAB calls in which `skip_mask.any()` was true: 0 of 3 as shipped, 3 of 3 with the fix. What the reserved slots actually contain on the way out of the shipped column stage is noise near the sentinel (range `-100.59 .. -99.38`), not `-100.0`.

Wasted work is `reserve_cls_tokens / (reserve_cls_tokens + G)` of stage-one compute, since the group axis is folded into the batch — ~44% on a 5-feature table, ~4% on a 100-feature one. End-to-end with the real `tabicl-classifier-v2-20260212.ckpt` checkpoint (`n_estimators=1`, same `T=1000` tables, 5–50 features) the fast path fires 0/12 times as shipped and 12/12 with the fix, the whole-model speedup on CPU is ~1.05x–1.1x up to 25 features and within timing noise at 50, and the logits are **bit-identical** (`max |Δlogit| = 0.0`), because `RowInteraction` overwrites `embeddings[:, :, :num_cls]` unconditionally in both forward paths (`_model/interaction.py:201`, `:248`).

### Scope

Paths I checked in 2.2.0 (line numbers identical in 2.1.1):

| path | status |
|---|---|
| `_compute_embeddings`, `target_aware=True`, standard branch (`embedding.py:389`) | fast path dead — this is the released config |
| `_compute_embeddings`, mixed-radix branch (`embedding.py:409`) | same pattern, same effect |
| `_train_forward_without_feature_group` with `d` given | routes through `_compute_embeddings`, same |
| `_compute_embeddings_with_cache`, `store_cache=True` (`embedding.py:789`) | sentinel also destroyed, but `forward_with_cache` (`layers.py:787`) deliberately keeps batch dims for the cache and only restores the value, so no FLOPs are lost there |
| `_compute_embeddings_with_cache`, `use_cache=True` | no `y_emb` addition, sentinel survives |
| `target_aware=False` | fires as intended |

So the fix belongs in `_compute_embeddings` (both branches); the cache path would only gain consistency, not speed, and the PR leaves it alone.

### Environment

tabicl 2.2.0 (first seen on 2.1.1) · torch 2.13.0 · Python 3.12.13 · macOS 26.6.2 (arm64), CPU
